packages/apollo-datasource-rest/src/HTTPCache.ts

```diff
@@ -0,0 +1,118 @@
+import CachePolicy from 'http-cache-semantics';
+
+import { KeyValueCache } from './KeyValueCache';
+import { InMemoryKeyValueCache } from './InMemoryKeyValueCache';
+
+export class HTTPCache {
+  constructor(
+    private keyValueCache: KeyValueCache = new InMemoryKeyValueCache(),
+  ) {}
+
+  async fetch(input: RequestInfo, init?: RequestInit): Promise<Response> {
+    const request = new Request(input, init);
+
+    const cacheKey = cacheKeyFor(request);
+
+    const entry = await this.keyValueCache.get(cacheKey);
+    if (!entry) {
+      const response = await fetch(request);
+
+      const policy = new CachePolicy(
+        policyRequestFrom(request),
+        policyResponseFrom(response),
+      );
+
+      return this.storeResponseAndReturnClone(request, response, policy);
+    }
+
+    const { policy: policyRaw, body } = JSON.parse(entry);
+
+    const policy = CachePolicy.fromObject(policyRaw);
+
+    if (policy.satisfiesWithoutRevalidation(policyRequestFrom(request))) {
+      const headers = policy.responseHeaders();
+      return new Response(body, { status: policy._status, headers });
```

Comment:
Instead of depending on this internal detail, just save `status` separately in our cache alongside `policy` and `body`.  (There's another case of it being used below; I'm pretty sure that revalidatedPolicy._status is always equal to policy._status so you can just use the same saved value.)